spyder/api/plugins/new_api.py

```diff
@@ -1092,327 +1664,557 @@ def __init__(self, parent, configuration):
+    def after_long_process(self, message: str = "") -> None:
         """
-        Clear main window's status bar after a long process, restore mouse
-        pointer to the OS deault and stop spinner.
+        Perform actions required after starting a longer-running process.
+
+        Clears the message in the Spyder main window's status bar,
+        restores the mouse pointer to the OS default, and stops spinner.
 
         Parameters
         ----------
-        message: str
+        message: str, optional
             Message to show in the status bar when the long process finishes.
+            An empty message by default, clearing the previous message.
+
+        Returns
+        -------
+        None
         """
         super().after_long_process(message)
         self.get_widget().stop_spinner()
 
     def get_widget(self) -> PluginMainWidget:
         """
-        Return the plugin main widget.
+        Return the plugin's main widget.
+
+        Returns
+        -------
+        spyder.api.widgets.main_widget.PluginMainWidget
+            The plugin's main widget, an instance of :attr:`WIDGET_CLASS`.
         """
         if self._widget is None:
-            raise SpyderAPIError('Dockable Plugin must have a WIDGET_CLASS!')
+            raise SpyderAPIError("Dockable Plugin must have a WIDGET_CLASS!")
 
         return self._widget
 
-    def update_title(self):
+    def update_title(self) -> None:
         """
-        Update plugin title, i.e. dockwidget or window title.
+        Update the plugin widget's title, i.e. its dockwidget or window title.
+
+        Returns
+        -------
+        None
         """
         self.get_widget().update_title()
 
-    def update_margins(self, margin):
+    def update_margins(self, margin: int | None) -> None:
         """
-        Update margins of main widget inside dockable plugin.
+        Update the margins of the main widget inside this dockable plugin.
+
+        Parameters
+        ----------
+        margin: int | None
+            The margins to use for the main widget, or ``None`` for the
+            default margins.
+
+        Returns
+        -------
+        None
         """
         self.get_widget().update_margins(margin)
 
     @Slot()
-    def switch_to_plugin(self, force_focus=False):
+    def switch_to_plugin(self, force_focus: bool = False) -> None:
         """
-        Switch to plugin and define if focus should be given or not.
+        Switch to this plugin and define if focus should be given or not.
+
+        Parameters
+        ----------
+        force_focus : bool | None, optional
+            If ``True``, always give the plugin's widget focus when showing
+            or hiding it with this method. If ``None``, only give it focus when
+            showing, not hiding (i.e. setting ``state`` to ``True``),
+            and only if :attr:`RAISE_AND_FOCUS` is ``True``. If ``False``,
+            the default, don't give it focus regardless.
```

Comment:
This docstring has the same problem as the one for its corresponding signal (see above).

Reply:
Accordingly:

```suggestion
            If ``True``, always give the plugin's widget focus when showing
            or hiding it with this method. If ``False``, the default,
            don't give it focus.
```